### PR TITLE
feat(Image): Added tint color, background color and blur radius.

### LINKED ISSUE #278: Fix remaining Image features
The following Image features are not working in UIExplorer:
- Resize Method
- Tint Color
- Background Color
- Plain Static Image

### PR DESCRIPTION
**Pull request by phongcao**
New features:

- Tint color and background color.
- Blur radius (Windows 10 Anniversary or newer).

Fix #278

### CODE REVIEW

**Comment by rozele** [comment]
@phongcao please rebase, I think there's still a CI issue, 🤞 hoping its fixed.

**Comment by phongcao** [comment]
@rozele Done.

File: ReactWindows/ReactNative/Views/Image/ReactImageManager.cs [hunk 1]
```diff
@@ -273,21 +320,36 @@ private void OnImageStatusUpdate(Border view, ImageLoadStatus status, ImageMetad
...
                 var image = default(BitmapSource);
                 var uri = new Uri(source);
+                var builder = ImageRequestBuilder.NewBuilderWithSource(uri);
+                var imageData = GetImageData(view);
 
-                // Remote images
-                if (source.StartsWith("http:") || source.StartsWith("https:"))
+                // Apply postprocessor if Composition isn't supported.
+                if (imageData.BackgroundColor != null || 
+                    (imageData.TintColor != null && !EffectManager.IsCompositionSupported))
```
Comment (rozele) [inline ReactWindows/ReactNative/Views/Image/ReactImageManager.cs]: It looks like the TintColor will always be applied by the postprocessor if a BackgroundColor is set, even if the Composition is enabled.

Comment (phongcao) [inline ReactWindows/ReactNative/Views/Image/ReactImageManager.cs]: Yes, because the background color can't be achieved using Composition API.

File: ReactWindows/ReactNative/Views/Image/ColorFilterPostprocessor.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        /// <param name="height">
+        /// The height of the new software bitmap, in pixels.
+        /// </param>
+        /// <param name="format">
+        /// The pixel format of the new software bitmap.
+        /// </param>
+        /// <param name="alpha">
+        /// The alpha mode of the new software bitmap.
+        /// </param>
+        public override void Process(
```
Comment (rozele) [inline ReactWindows/ReactNative/Views/Image/ColorFilterPostprocessor.cs]: I'm not certain this background color filter is quite what we need. I don't think it deals with modifying partially transparent pixels.

Since the composition approaches work very well, maybe we should put this "patch" in a separate PR, so we can revert / decide on them separately if necessary.

File: ReactWindows/ReactNative/Views/Image/ReactImageData.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using Windows.UI.Xaml;
+
+namespace ReactNative.Views.Image
+{
+    class ReactImageData : IDisposable
```
Comment (rozele) [inline ReactWindows/ReactNative/Views/Image/ReactImageData.cs]: >ReactImageData [](start = 10, length = 14)

ReactImageManager is already a monolith. It probably wouldn't hurt for this class to be a nested class in ReactImageManager.

Also if possible, it would be good if we keep this as a "properties-only" class, that way the side-effects of the ViewManager lifecycle events and prop hooks are much more explicit.

File: ReactWindows/ReactNative/Views/Image/ReactImageManager.cs (hunk 1 above)
Comment (phongcao) [inline ReactWindows/ReactNative/Views/Image/ReactImageManager.cs]: Nope, it's ignored by the postprocessor if it's null.

https://github.com/Microsoft/react-native-windows/pull/1319/files#diff-1411f606f43af4e7125cf4ab3d01d1fb

File: ReactWindows/ReactNative/Views/Image/ColorFilterPostprocessor.cs (hunk 2 above)
Comment (phongcao) [inline ReactWindows/ReactNative/Views/Image/ColorFilterPostprocessor.cs]: Do you have any example about the partially transparent pixels? Agree that this should be put in a different PR so that we can revert easily.

**Comment by phongcao** [comment]
Closing this one and will create two other PRs to address the feedback.
